src/main/scala/rocketchip/BaseTop.scala

```diff
@@ -67,7 +67,8 @@ abstract class BaseTopModule[+L <: BaseTop, +B <: BaseTopBundle](
   val io: B = b
 
   val coreplex = p(BuildCoreplex)(outer.c, p)
-  val coreplexIO = coreplex.io
+  val coreplexIO = Wire(new BaseCoreplexBundle(outer.c)(p){})
```

Comment:
Actually, similar to ParameterizedBundle, you can do this in the BaseCoreplexBundle, and get rid of all cloneTypes in the extended classes:
```
  override def cloneType: BaseCoreplexBundle.this.type =
    this.getClass.getConstructors.head.newInstance(c, p).asInstanceOf[this.type]
```